lib/setup_azure_keyvault.sh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+libroot="$(dirname "$(readlink -f "$0")")"
+
+if [[ $# -eq 0 || $# -ge 3 ]]; then
+  echo "Usage: $0 <RG name> [<AZURE_SUBSCRIPTION_ID>]"
+  echo "Note: AZURE_SUBSCRIPTION_ID can also be specified via environment variable"
+  exit 1
+fi
+
+set +u
+AZURE_SUBSCRIPTION_ID="${AZURE_SUBSCRIPTION_ID:-$2}"
+set -u
+RG_NAME="$1"
+LOCATION="westeurope"
+KV_NAME="sysext-bakery"
+CERT_NAME="sysext-bakery-signing"
+IDENTITY_NAME="sysext-bakery-github-actions"
+FED_CRED_NAME="gh-main"
+GITHUB_REPO=$(source "${libroot}/libbakery.sh"; echo "$bakery") # owner/repo
+GITHUB_BRANCH="main"
+
+echo "Setting subscription..."
+az account set --subscription "$AZURE_SUBSCRIPTION_ID"
+TENANT_ID="$(az account show --query tenantId -o tsv)"
+
+echo "Creating resource group $RG_NAME in $LOCATION..."
+az group create -n "$RG_NAME" -l "$LOCATION" -o none
+
+# ============
+# KEY VAULT
+# ============
+echo "Creating Key Vault $KV_NAME..."
+# Enable RBAC authorization and purge protection (soft-delete is on by default)
+#
+az keyvault show -n "$KV_NAME" -g "$RG_NAME" -o none 2>/dev/null || \
+  az keyvault create \
+    --name "$KV_NAME" \
+    --resource-group "$RG_NAME" \
+    --location "$LOCATION" \
+    --sku standard \
+    --enable-purge-protection true \
+    --enable-rbac-authorization true \
+    -o none
+KV_ID="$(az keyvault show -n "$KV_NAME" -g "$RG_NAME" --query id -o tsv)"
+
+# 2) Get your Azure AD object id (the caller shown in the error)
+#    This works when you're logged in as a user (not a service principal)
+USER_OID=$(az ad signed-in-user show --query id -o tsv)
+USER_NAME=$(az ad signed-in-user show --query userPrincipalName -o tsv)
+RG_ID=$(az group show -n "$RG_NAME" --query id -o tsv)
+
+# 3) Grant yourself a data-plane role that includes certificate create
+#    Pick ONE of the two; Administrator is broader than Certificates Officer.
+for role in "Key Vault Administrator" "Owner"; do
+  echo "Granting $role role in RG $RG_NAME to $USER_NAME"
+  az role assignment create \
+    --assignee-object-id "$USER_OID" \
+    --assignee-principal-type User \
+    --role "$role" \
+    --scope "$RG_ID" \
+    -o none
+done
+
+echo "Waiting 30 seconds for the permissions to apply. If the script fails, please wait and run it again."
+sleep 30
```

Comment:
Do we have any chance to poll for the completion instead of defining a wait time that may be exceeded?